crates/ibc/src/core/ics02_client/events.rs

```diff
@@ -390,3 +390,101 @@ impl From<UpgradeClient> for abci::Event {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::mock::client_state::{client_type as mock_client_type, MOCK_CLIENT_TYPE};
+    use crate::mock::header::MockHeader;
+    use ibc_proto::google::protobuf::Any;
+    use tendermint::abci::Event as AbciEvent;
+
+    #[test]
+    fn ibc_to_abci_client_events() {
+        struct Test {
+            kind: IbcEventType,
+            event: AbciEvent,
+            expected_keys: Vec<&'static str>,
+            expected_values: Vec<String>,
+        }
+
+        let client_type = mock_client_type();
+        let client_id = ClientId::new(client_type.clone(), 0).unwrap();
+        let consensus_height = Height::new(0, 10).unwrap();
+        let consensus_heights = vec![Height::new(0, 5).unwrap(), Height::new(0, 7).unwrap()];
+        let header: Any = MockHeader::new(consensus_height).into();
+        let expected_keys = vec![
+            CLIENT_ID_ATTRIBUTE_KEY,
+            CLIENT_TYPE_ATTRIBUTE_KEY,
+            CONSENSUS_HEIGHT_ATTRIBUTE_KEY,
+            CONSENSUS_HEIGHTS_ATTRIBUTE_KEY,
+            HEADER_ATTRIBUTE_KEY,
+        ];
+        let expected_values = vec![
+            client_id.to_string(),
+            MOCK_CLIENT_TYPE.to_string(),
+            consensus_height.to_string(),
+            consensus_heights
+                .iter()
+                .map(|h| h.to_string())
+                .collect::<Vec<_>>()
+                .join(","),
+            String::from_utf8(hex::encode(header.clone().value)).unwrap(),
+        ];
```

Comment:
Similarly here, can we use the correct hardcoded values?

Reply:
Same comment applies to the other 2 `events.rs` files

Reply:
https://github.com/cosmos/ibc-rs/pull/363/commits/ffa0bbcf972b93e074829390cb9cd3511eae23f9